astropy/cosmology/io/html.py

```diff
@@ -0,0 +1,103 @@
+        the value of that column. If the table is not indexed and ``index``
+        is a string, the "name" column is used as the indexing column.
+
+    move_to_meta : bool (optional, keyword-only)
+        Whether to move keyword arguments that are not in the Cosmology class'
+        signature to the Cosmology's metadata. This will only be applied if the
+        Cosmology does NOT have a keyword-only argument (e.g. ``**kwargs``).
+        Arguments moved to the metadata will be merged with existing metadata,
+        preferring specified metadata in the case of a merge conflict
+        (e.g. for ``Cosmology(meta={'key':10}, key=42)``, the ``Cosmology.meta``
+        will be ``{'key': 10}``).
+
+    cosmology : str, `~astropy.cosmology.Cosmology` class, or None (optional, keyword-only)
+        The cosmology class (or string name thereof) to use when constructing
+        the cosmology instance. The class also provides default parameter values,
+        filling in any non-mandatory arguments missing in 'table'.
+
+    **kwargs
+        Passed to :attr:`astropy.table.QTable.read`
+
+    Returns
+    -------
+    `~astropy.cosmology.Cosmology` subclass instance
+    """
+    kwargs["format"] = "ascii.html"
+    with u.add_enabled_units(cu):
+        table = QTable.read(filename, **kwargs)
+
+    # build cosmology from table
+    return from_table(table, index=index, move_to_meta=move_to_meta, cosmology=cosmology)
+
+def write_table(cosmology, file, *, overwrite=False, cls=QTable, format_dict=None, cosmology_in_meta=True, **kwargs):
+    """Serialize the cosmology into a HTML table.
+
+    Parameters
+    ----------
+    cosmology : `~astropy.cosmology.Cosmology` subclass instance
+    file : path-like or file-like
+        Location to save the serialized cosmology.
+
+    overwrite : bool
+        Whether to overwrite the file, if it exists.
+    cls : type (optional, keyword-only)
+        Astropy :class:`~astropy.table.Table` (sub)class to use when writing.
+        Default is :class:`~astropy.table.QTable`.
+    cosmology_in_meta : bool
+        Whether to put the cosmology class in the Table metadata (if `True`,
+        default) or as the first column (if `False`).
+    **kwargs
+        Passed to ``cls.write``
+
+    Raises
+    ------
+    TypeError
+        If kwarg (optional) 'cls' is not a subclass of `astropy.table.Table`
+    """
+    table = to_table(cosmology, cls, cosmology_in_meta=False)
+
+    if cls == QTable:
+        cosmo_cls = type(Planck18)
+        for n, col in zip(table.colnames, table.itercols()):
+            if n != "cosmology":
+                param = getattr(cosmo_cls, n)  # Get parameter, and check it; fails here
+                if not isinstance(param, Parameter) or param.unit in (None, u.one):
+                    continue
+                # Replace column with unitless version
+                table.replace_column(n, col.to_value(param.unit), copy=False)

```

Comment:
Nifty trick to reduce number of tabs and keep control flow easier to read:

```suggestion
            if n == "cosmology":
                continue
            param = getattr(cosmo_cls, n)  # Get parameter, and check it; fails here
            if not isinstance(param, Parameter) or param.unit in (None, u.one):
                continue
            # Replace column with unitless version
            table.replace_column(n, col.to_value(param.unit), copy=False)
```

Reply:
Thanks! Fixed in local version, will upload updated with the rest.